src/performance.js

```diff
@@ -58,8 +66,48 @@ export class Performance {
 
     /** @const @private {!Array<TickEventDef>} */
     this.events_ = [];
+
+    /** @private {boolean} */
+    this.isViewerVisible_ = false;
+
+    this.win.addEventListener('unload', this.flush.bind(this));
   }
 
+  /**
+   * Gets primary amp core services.
+   * @return {!Performance}
+   */
+  init() {
```

Comment:
honestly, did this just to make mocking them easier. had a hard time mocking the viewer_ and resources_ when real services where in test. can you point me to some test files where we mocked these?